h/services/profile_group.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+
+
+class ProfileGroupService(object):
+
+    """A service for retrieving groups associated with users."""
+
+    def __init__(self, session, open_group_finder):
+        """
+        Create a new profile_groups service.
+
+        :param session: the SQLAlchemy session object
+        :param open_group_finder: a callable for finding open groups
+        """
+        self.session = session
+        self.open_group_finder = open_group_finder
+
+    def all(self, user=None, authority=None):
+        """ return a list of all user groups"""
+
+        open_groups = self.open_group_finder(authority) or []
+        private_groups = []
+        if user is not None:
+            private_groups = user.groups
+
+        groups = open_groups + private_groups
```

Comment:
Ok, so this would be all the open groups (of the same authority) in Hypothesis (since all users can read all open groups in their authority) plus the user's private groups. Is the _Public_ group in this list?

Reply:
(I think the _Public_ group is in there, in the `open_groups` list)

Reply:
Yes, it would be (the _Public_ group). This part of the service, as well as the needs-to-be-renamed `authority_group` service, will need refactoring and review as we continue development—this isn't the end point of where this will end up, I'm sure.